src/js/events/MesosSummaryActions.js

```diff
@@ -17,7 +17,10 @@ var _historyServiceOnline = true;
 function testHistoryServerResponse(response) {
   // If the response is an empty object, that means something is whack
   // Fall back to making requests to Mesos
-  if (response === {}) {
+  // TODO (DCOS-7764): This should be improved to validate against a schema
+  if (response === {} ||
+      Array.isArray(response.frameworks) ||
```

Comment:
Shouldn't this be `!Array.isArray(response.frameworks)` and `Array.isArray(response.slaves)`?